lib/Sema/MiscDiagnostics.cpp

```diff
@@ -3332,7 +3332,22 @@ class ObjCSelectorWalker : public ASTWalker {
           name = bestMethod->getFullName();
         }
 
-        out << nominal->getName().str() << "." << name.getBaseName();
+        auto typeName = nominal->getName().str();
+        // If we're inside a type Foo (or an extension of it) and the suggestion
+        // is going to be #selector(Foo.bar) (or #selector(SuperclassOfFoo.bar),
+        // then suggest the more natural #selector(self.bar) instead.
+        if (auto containingTypeContext = DC->getInnermostTypeContext()) {
+          auto methodNominalType = nominal->getDeclaredType();
```

Comment:
Could any of these ```auto``` be ```const```

Reply:
Please stop "reviewing" random inactive PRs. If you want to participate, look at what's going on now.